app/src/main/java/app/revanced/integrations/youtube/patches/components/HideKeywordContentFilter.java

```diff
@@ -0,0 +1,137 @@
+     */
+    private final StringFilterGroup containsFilter = new StringFilterGroup(
+            Settings.HIDE_KEYWORD_CONTENT,
+            "modern_type_shelf_header_content.eml");
+
+    private final ByteTrieSearch bufferSearch = new ByteTrieSearch();
+
+    /**
+     * Minimum keyword/phrase length to prevent excessively broad content filtering.
+     */
+    private static final int MINIMUM_KEYWORD_LENGTH = 3;
+
+    private static List<String> splitAndVerifyStrings(StringSetting setting) {
+        String[] split = setting.get().split("\n");
+        List<String> keywords = new ArrayList<>(split.length);
+
+        for (String phrase : split) {
+            // Remove any trailing white space the user may have accidentally included.
+            phrase = phrase.stripTrailing();
+            if (phrase.isBlank()) continue;
+
+            if (phrase.length() < MINIMUM_KEYWORD_LENGTH) {
+                Utils.showToastLong(str("revanced_hide_keyword_toast_invalid_length", MINIMUM_KEYWORD_LENGTH, phrase));
+                Utils.showToastLong(str("revanced_hide_keyword_toast_reset"));
+                setting.resetToDefault();
+                return splitAndVerifyStrings(setting);
+            }
+            keywords.add(phrase);
+        }
+
+        return keywords;
+    }
+
+    private void parseKeywords() {
+        List<String> keywords = splitAndVerifyStrings(Settings.HIDE_KEYWORD_CONTENT_PHRASES);
+        if (keywords.isEmpty()) {
+            return;
+        }
+
+        // Add common casing that might appear.
+        // If a desired keyword has mixed casing (such as "Matt Whatever")
+        // the user will need to add that exact string.
+        // This could be simplified by adding case insensitive search to the prefix search,
+        // but that also brings a small performance hit.
+        List<String> modifiedKeywords = new ArrayList<>();
+        for (String keyword : keywords) {
+            modifiedKeywords.add(keyword); // Original casing as added by the user
+
+            // Add lower case and upper case variants.
+            String lowerCase = keyword.toLowerCase();
+            if (!keyword.equals(lowerCase)) {
+                modifiedKeywords.add(lowerCase);
+            }
+            String upperCase = keyword.toUpperCase();
+            if (!keyword.equals(upperCase)) {
+                modifiedKeywords.add(upperCase);
+            }
+            // Include first letter capitalization variant.
+            String sentenceCapital = Character.toUpperCase(keyword.charAt(0))
+                    + keyword.substring(1);
+            if (!sentenceCapital.equals(keyword) && !sentenceCapital.equals(upperCase)) {
+                modifiedKeywords.add(sentenceCapital);
+            }
+        }
+        Logger.printDebug(() -> "Using keywords: " + modifiedKeywords);
+        bufferSearch.addPatterns(convertStringsToBytes(modifiedKeywords.toArray(new String[0])));
```

Comment:
Duplicates may be filtered here. The user may try enter casings that may have been added by the hardcoded implementation already.